packages/react/src/chip/Chip.tsx

```diff
@@ -0,0 +1,60 @@
+import * as RadixToggle from "@radix-ui/react-toggle";
+import { type ComponentPropsWithRef, type ReactNode, forwardRef } from "react";
+
+import type { ExtendProps } from "../utils";
+
+import { Box } from "../box";
+import * as styles from "./Chip.css";
+
+type ChipProps = ExtendProps<
+  ComponentPropsWithRef<typeof RadixToggle.Root>,
+  ComponentPropsWithRef<typeof Box>,
+  {
+    children?: ReactNode;
+    disabled?: boolean;
+    endDecorator?: ReactNode;
+    startDecorator?: ReactNode;
+  } & styles.ChipVariants
+>;
```

Comment:
We can now simplify this to:

```suggestion
type ChipProps = BoxProps<
  typeof RadixToggle.Root,
  {
    endDecorator?: ReactNode;
    startDecorator?: ReactNode;
  } & styles.ChipVariants
>;
```

Reply:
Addressed bhai